exchange/utils_test.go

```diff
@@ -2228,3 +2228,98 @@ func TestRemoveUnpermissionedEidsEmptyValidations(t *testing.T) {
 		assert.Equal(t, &requestExpected, test.request, test.description+":request")
 	}
 }
+
+func TestBuildXPrebidHeader(t *testing.T) {
+	testCases := []struct {
+		description   string
+		version       string
+		requestExt    *openrtb_ext.ExtRequest
+		requestAppExt *openrtb_ext.ExtApp
+		result        string
+	}{
+		{
+			description: "No versions",
+			version:     "",
+			result:      "pbs-go/unknown",
+		},
+		{
+			description: "pbs",
+			version:     "test-version",
+			result:      "pbs-go/test-version",
+		},
+		{
+			description: "prebid.js",
+			version:     "test-version",
+			requestExt: &openrtb_ext.ExtRequest{
+				Prebid: openrtb_ext.ExtRequestPrebid{
+					Channel: &openrtb_ext.ExtRequestPrebidChannel{
+						Name:    "pbjs",
+						Version: "test-pbjs-version",
+					},
+				},
+			},
+			result: "pbs-go/test-version,pbjs/test-pbjs-version",
+		},
+		{
+			description: "unknown prebid.js",
+			version:     "test-version",
+			requestExt: &openrtb_ext.ExtRequest{
+				Prebid: openrtb_ext.ExtRequestPrebid{
+					Channel: &openrtb_ext.ExtRequestPrebidChannel{
+						Name: "pbjs",
+					},
+				},
+			},
+			result: "pbs-go/test-version,pbjs/unknown",
+		},
+		{
+			description: "channel without a name",
+			version:     "test-version",
+			requestExt: &openrtb_ext.ExtRequest{
+				Prebid: openrtb_ext.ExtRequestPrebid{
+					Channel: &openrtb_ext.ExtRequestPrebidChannel{
+						Version: "test-version",
+					},
+				},
+			},
+			result: "pbs-go/test-version",
+		},
+		{
+			description: "prebid-mobile",
+			version:     "test-version",
+			requestAppExt: &openrtb_ext.ExtApp{
+				Prebid: openrtb_ext.ExtAppPrebid{
+					Source:  "prebid-mobile",
+					Version: "test-prebid-mobile-version",
+				},
+			},
+			result: "pbs-go/test-version,prebid-mobile/test-prebid-mobile-version",
+		},
+		{
+			description: "app ext without a source",
+			version:     "test-version",
+			requestAppExt: &openrtb_ext.ExtApp{
+				Prebid: openrtb_ext.ExtAppPrebid{
+					Version: "test-version",
+				},
+			},
+			result: "pbs-go/test-version",
+		},
```

Comment:
Can we add a test where both `requestExt` and `requestAppExt` hold valid values?
```
2232 func TestBuildXPrebidHeader(t *testing.T) {
2233     testCases := []struct {
2234         description   string
2235         version       string
2236         requestExt    *openrtb_ext.ExtRequest
2237         requestAppExt *openrtb_ext.ExtApp
2238         result        string
2239     }{
2240         {
2241             description: "No versions",
2242             version:     "",
2243             result:      "pbs-go/unknown",
2244         },
2245 -- 53 lines: {------------------------------------------------------------
2298         {
2299             description: "app ext without a source",
2300             version:     "test-version",
2301             requestAppExt: &openrtb_ext.ExtApp{
2302                 Prebid: openrtb_ext.ExtAppPrebid{
2303                     Version: "test-version",
2304                 },
2305             },
2306             result: "pbs-go/test-version",
2307         },
     +       {
     +           description: "Version found in both req.Ext and req.App.Ext",
     +           version:     "test-version",
     +           requestExt: &openrtb_ext.ExtRequest{  // Valid
     +               Prebid: openrtb_ext.ExtRequestPrebid{
     +                   Channel: &openrtb_ext.ExtRequestPrebidChannel{
     +                       Name:    "pbjs",
     +                       Version: "test-pbjs-version",
     +                   },
     +               },
     +           },
     +           requestAppExt: &openrtb_ext.ExtApp{  // Valid
     +               Prebid: openrtb_ext.ExtAppPrebid{
     +                   Source:  "prebid-mobile",
     +                   Version: "test-prebid-mobile-version",
     +               },
     +           },
     +           result: "PUT BOTH VERSIONS IN THE RESULT OR ONLY ONE OF THEM, WHICHEVER TAKES PRIORITY"
     +       },
2308     }
2309
2310     for _, test := range testCases {
2311         req := &openrtb2.BidRequest{}
2312 -- 11 lines: if test.requestExt != nil {--------------------------------
2323         assert.Equal(t, test.result, result, test.description+":result")
2324     }
2325 }
exchange/utils_test.go
```

Reply:
Done.